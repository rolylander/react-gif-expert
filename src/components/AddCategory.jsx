import { useState } from "react"


export const AddCategory = ({onNewCategory}) => {    
    const [inputValue, setInputValue] = useState('')
    const onInputChange = ({ target }) => {
        
        setInputValue(target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault()
        const _inputValue = inputValue.trim()
        if(_inputValue.length<=1) return
        
        onNewCategory(_inputValue)
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit}>
            <input type="text" placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>

    )
}
