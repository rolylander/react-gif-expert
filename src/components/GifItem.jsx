import PropTypes from 'prop-types'

export const GifItem = ({title,url,id}) => {

    
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

/*
1. aniadir proptypes yarn add
- title obligatorio
- url obligatorio

2. evaluar snapshot

*/
