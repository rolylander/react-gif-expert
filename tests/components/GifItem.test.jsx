import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"




describe('prueba <GiftItem/>', () => {
    const title = "saitama"
    const url = "https://giphy.com/gifs/playvalorant-uBRwFMpdukW1hf8yqS"

    test('debe mandar titulo y url', () => {
        render(<GifItem title={title} url={url} />)
    })
    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)

    })
    test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()

     })

})





/*
1. aniadir proptypes yarn add
- title obligatorio
- url obligatorio

2. evaluar snapshot

*/