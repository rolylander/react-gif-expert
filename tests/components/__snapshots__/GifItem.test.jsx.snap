// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba <GiftItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://giphy.com/gifs/playvalorant-uBRwFMpdukW1hf8yqS"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
